src/Dapr.Client/BulkStateItem.cs

```diff
@@ -49,4 +49,41 @@ public BulkStateItem(string key, string value, string etag)
         /// </summary>
         public string ETag { get; }
     }
+
+    /// <summary>
+    /// Represents a state object returned from a bulk get state operation.
+    /// </summary>
+    public readonly struct BulkStateItem<TValue>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BulkStateItem"/> class.
+        /// </summary>
+        /// <param name="key">The state key.</param>
+        /// <param name="value">The value.</param>
+        /// <param name="etag">The ETag.</param>
+        /// <remarks>
+        /// Application code should not need to create instances of <see cref="BulkStateItem" />.
+        /// </remarks>
+        public BulkStateItem(string key, TValue value, string etag)
+        {
+            this.Key = key;
+            this.Value = value;
+            this.ETag = etag;
+        }
+
+        /// <summary>
+        /// Gets the state key.
+        /// </summary>
+        public string Key { get; }
+
+        /// <summary>
+        /// Gets the value.
+        /// </summary>
+        public TValue Value { get; }
```

Comment:
I might add remarks to the original `BulkStateItem.Value` docs about it being (possibly) serialized and might need deserialization.

Reply:
Updated to provide a little more context here.